### ISSUE TITLE
Fill in spacer-repeat join table and repeat-spacer-organism table in database

### ISSUE THREAD

**Issue by mbonsma** [goyalsid/phageParser#104]
After Issue #99, we'll have the tables **Organism**, **Spacer**, and **Repeat**. The interesting part is to make a record of the connections between all three. In an actual CRISPR system, an organism may have zero, one, or more than one CRISPR locus, and each locus will (usually) have several copies of the same repeat and two or more associated spacers (see the [README](https://github.com/goyalsid/phageParser/blob/master/README.md#about-the-crispr-system) for more info!). 

We have two join tables: **SpacerRepeatPair** and **OrganismSpacerRepeatPair** that have these columns respectively:

| SpacerRepeatPairID | SpacerID | RepeatID |
| --- | --- | --- |

| OrganismSpacerRepeatPairID | OrganismID | SpacerRepeatPairID | SpacerRepeatPairLeftID |
| --- | --- | --- | --- |
### The function

**Input:**
- [database file](https://github.com/goyalsid/phageParser/blob/master/crispr.sqlite)
- list of CRISPR repeats with organism accession numbers from CRISPRdb: 
  [online here](http://crispr.i2bc.paris-saclay.fr/crispr/BLAST/DR/DRdatabase) or [on GitHub here](https://github.com/goyalsid/phageParser/blob/master/data/DRdatabase.txt)
- list of CRISPR spacers with organism accession numbers from CRISPRdb: 
  [online here](http://crispr.i2bc.paris-saclay.fr/crispr/BLAST/Spacer/Spacerdatabase) or [on GitHub here](https://github.com/goyalsid/phageParser/blob/master/data/spacerdatabase.txt)

**The function should do the following:**
- Part 1: for each spacer in the **Spacer** table, find its associated repeat using the data from CRISPRdb and fill in their primary keys in the **SpacerRepeatPair** table. I'm not sure what the best way to do this is, suggestions welcome!
- Part 2 (the weirder part): for each spacer-repeat pair in the _SpacerRepeatPair_\* table, find the associated organism accession number (i.e. `NC_000853`) and enter that organism primary key (from the OrganismID table) in the **OrganismSpacerRepeatPair** table. The last column, **SpacerRepeatPairLeftID**, is the primary key from the **SpacerRepeatPair** table for the **spacer that comes before the current spacer**. 

**Example (part 2):**

Here's what the tables could look like before starting:

| OrganismID | OrganismName | Accession |
| --- | --- | --- |
| 1 | Thermotoga maritima MSB8 | NC_000853 |

| SpacerID | SpacerSequence |
| --- | --- |
| 1 | TCTGTGATCTGTTAACGGGGACATCCACACCCAAAATC |
| 2 | ACTGTCGGAGATGTTCTGCTGTGCATCGTCGAGAGTTT |

| RepeatID | RepeatSequence |
| --- | --- |
| 1 | GTTTCAATAATTCCTTAGAGGTATGGAAAC |

| SpacerRepeatPairID | SpacerID | RepeatID |
| --- | --- | --- |
| 1 | 1 | 1 |
| 1 | 2 | 1 |

Now we want to figure out how to fill in the **OrganismSpacerRepeatPair** table. Here are the first few lines of the repeats file and spacers file, respectively:

```
>NC_000853_1|NC_000853_2|NC_000853_3|NC_000853_4|NC_000853_5|NC_000853_8|NC_009486_4|NC_009486_6|NC_009486_8|NC_009486_9|NC_010483_4|NC_010483_7|NC_010483_8|NC_013642_3|NC_013642_6|NC_013642_7|NC_013642_8|NC_023151_1|NC_023151_2|NC_023151_3|NC_023151_4|NC_023151_5|NC_023151_7
GTTTCAATAATTCCTTAGAGGTATGGAAAC
```

```
>NC_000853_1_1|NC_023151_4_41
TCTGTGATCTGTTAACGGGGACATCCACACCCAAAATC
>NC_000853_1_10|NC_023151_4_32
TATACGTTGTAGAAATCGCAAAGGTGGTGATGTTA
>NC_000853_1_11|NC_023151_4_30
TCTCGAATCCGATATAATCATATTTCCTTATTATAT
>NC_000853_1_12|NC_023151_4_29
CCGCATCTCGTAGGTATTGCTTGCCTTCTGTTCCCTT
>NC_000853_1_13|NC_023151_4_28
TAAACTTGTAAATCCTTTTGAAGACTTAGAACAAC
>NC_000853_1_14|NC_023151_4_27
GAAACGCCCGCAAACGCTCATCACAAGCCCGTTTCC
>NC_000853_1_15|NC_023151_4_26
TTGAGTATAAAATCTTCTTCTGATAATGACAAAACA
>NC_000853_1_16|NC_023151_4_25
TGCTTCTATTACTCTTTGTAGCACGTTCGTGAACTCA
>NC_000853_1_17|NC_023151_4_24
TCGATAATCTCAGATGCCTCGCCTCAATTTCAATTG
>NC_000853_1_18|NC_023151_4_23|NC_023151_4_31
CTCCACAATGAAAAGAATAGATGGTAAGGTGTGCTT
>NC_000853_1_19|NC_023151_4_22
CTACTATCTTCTATATATATATTTTTGTTTTTGTTTT
>NC_000853_1_2|NC_023151_4_40
ACTGTCGGAGATGTTCTGCTGTGCATCGTCGAGAGTTT
```

The spacers we want to place both come from organism `NC_000853`. They are spacers 1 and 2 respectively (represented in the file by `NC_000853_1_1` and `NC_000853_1_2`). The associated repeat is `NC_000853_1` - CRISPR locus 1 in organism `NC_000853`.

**Comment by Ky5595** [comment]
I would like to give it a try.
Is it still open ?

**Comment by mbonsma** [comment]
Hi @Ky5595, yes this one hasn't been done yet.

**Comment by mbonsma** [comment]
Just noticed that this has been resolved in PR #105.
